From 9f8fe8802cf1ab46983607e556b0a92b3207fef5 Mon Sep 17 00:00:00 2001
From: Ben Shelton <benjamin.h.shelton@intel.com>
Date: Fri, 8 Jun 2018 18:36:42 +0300
Subject: [PATCH] Backport 'ice-linux: Fix IDC ADK TX path'

This fixes an issue where the ADK Netdriver Tx path fails.

Change-Id: Ic12288db82d543c71eb742da3b16fe68a6e6f033
Signed-off-by: Ben Shelton <benjamin.h.shelton@intel.com>
---
 drivers/net/ethernet/intel/ice_sw/ice_idc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/intel/ice_sw/ice_idc.c b/drivers/net/ethernet/intel/ice_sw/ice_idc.c
index 62da73e..275ae79 100644
--- a/drivers/net/ethernet/intel/ice_sw/ice_idc.c
+++ b/drivers/net/ethernet/intel/ice_sw/ice_idc.c
@@ -729,7 +729,7 @@ ice_adk_strt_xmit_internal(struct ice_peer_dev __maybe_unused *peer_dev,
 	tx_ring = vsi->tx_rings[skb->queue_mapping];
 	tx_ring->ptp_ts_ena = txmd->ts_ena;
 	tx_ring->ptp_ts_idx = txmd->ts_index;
-	return ice_start_xmit(skb, skb->dev);
+	return ice_start_xmit(skb, netdev);
 }
 
 /**
-- 
2.7.4

